giraf/binaryGiraf/helper.go

```diff
@@ -0,0 +1,86 @@
+package binaryGiraf
+
+import (
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/giraf"
+	"github.com/vertgenlab/gonomics/simpleGraph"
+	"io"
+	"sync"
+)
+
+// Read will input a giraf.fe file and will decompress into a slice of giraf.Giraf.
+// Requires the graph (.gg) file used to generate the giraf for decompression
+func Read(giraffeFile string, graphFile string) []giraf.Giraf {
+	var answer []giraf.Giraf
+	file := fileio.EasyOpen(giraffeFile)
+	graph := simpleGraph.Read(graphFile)
+	defer file.Close()
+	reader := NewBinReader(file.BuffReader)
+	var curr giraf.Giraf
+	var err error
+	for curr, err = reader.Read(graph); err != io.EOF; curr, err = reader.Read(graph) {
+		common.ExitIfError(err)
+		answer = append(answer, curr)
+	}
+
+	// Close reader
+	err = reader.bg.Close()
+	common.ExitIfError(err)
+
+	return answer
+}
+
+// ReadToChan will input a fileio.EasyReader and decompress girafs in the file into a stream (channel) of giraf.Giraf.
+// ReadToChan is designed to be run as a goroutine. See GoReadToChan for internally handled goroutines.
+func ReadToChan(file *fileio.EasyReader, graph *simpleGraph.SimpleGraph, data chan<- giraf.Giraf, wg *sync.WaitGroup) {
+	var curr giraf.Giraf
+	var err error
+	reader := NewBinReader(file.BuffReader)
+	for curr, err = reader.Read(graph); err != io.EOF; curr, err = reader.Read(graph) {
```

Comment:
I think is is slightly inefficient because you have to check err twice (once for io.EOF) then again for nil or not nil. In most cases this will be nil so depending on how you set up your checks, you can actually get away with checking the err only once. The speed up is not life changing, but something to keep in mind is that even though logic statements are equivalent, the order of operations does make a small difference on performance.